src/dune_rules/pkg_rules.ml

```diff
@@ -1175,13 +1189,40 @@ end = struct
           (Context_name.build_dir ctx)
           (Dune_pkg.Lock_dir.Pkg.files_dir info.name ~lock_dir)
       in
+      let* inherited_post_depends =
+        (* Deduplicate package names, preserving the association
+           between package names and (any) one of the locations of the
+           package name (for error messages). *)
+        let all_locs_by_name =
+          List.concat_map depends ~f:(fun (pkg : Pkg.t) ->
```

Comment:
This step looks wrong to me. You should be traversing the post deps when computing the closure

Reply:
Are you suggesting to compute post deps when computing the `depends` field, just above this in the file? I deliberately separated the logic for computing inherited post deps from computing regular deps for clarity. Also note that we can't traverse post dependencies in the same way that we traverse regular dependencies as that would lead to cycles.

Why is it necessary to traverse post deps when computing the dependency closure?

Reply:
I mentioned that it was "wrong" before, but that is probably too strong of a statement. I rather meant "not obviously correct". I do not yet see the gain in clarity from splitting the computation of the closure into two steps, but I do understand why you've needed to do so to avoid the resolution cycle. It's not the most direct way to solve this problem though.

If you look at lib.ml, you will see that all of these problems have already been solved for libraries. In particular, post dependencies are analogous to dependencies marked with `re_exports`. My initial instinct is to solve the exact same problem in the exact same way, rather than introducing an approach with its own pros and cons.

I can take a look at doing this, since I'm not sure how familiar you are with lib.ml